### PR TITLE
New traitor item: Polarized Contacts

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
Adds a new simple but sweet traitor item for 3 tc, polarized contacts.
Use them in hand to upgrade your eyes to have +tier 2 flash resistance. Single use.
Now if you have welder protection of 2 or higher, and someone checks your eyes with a flashlight, your pupils to not dilate. This is potentially possible non-antagonistically through the use of upgraded robotic eyes.
## Why It's Good For The Game
The idea behind this change is to allow traitors some counterplay against revolutionaries and security. If a traitor thinks they are about to go into combat with people with flashes, they will be rewarded for preparing. It also fills a similar role as no-slip shoes as being a low cost but always nice to have passive item, which I don't think we have enough of.
Its arguable that this should only be worth 2 tc to be on-par with noslips, or that it should be more because flash protection is more valuable. 
## Changelog
:cl: itseasytosee
add: A new traitor item has been added, Polarized Contacts!
/:cl:

### CODE REVIEW

**Comment by Livrah** [comment]
Can it be seen via health analyzer / eyes removal?

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -397,6 +397,29 @@
...
+	to_chat(user, span_notice("You begin applying the contact lenses to your eyes..."))
+	if(do_after(user, 30))
+		if(eyes)
+			to_chat(user, span_notice("the contacts seamlessly merge with your iris."))
+			eyes.flash_protect = FLASH_PROTECTION_WELDER
+			to_chat(user, span_warning("The [src] disintegrates into nothing."))
+			qdel(src)
+		else
+			to_chat(user, span_warning("You have no eyes to apply the contacts to!"))
+
```
Comment (Sealed101) [inline code/modules/clothing/glasses/_glasses.dm:422]: early returns and seconds defines

we can hopefully use the `target` argument of the `do_after` to cancel our progress bar when the eyes are removed, as it checks for the target's loc staying the same. so long as eyes register the player's head as their loc it should be fine (?)

```suggestion
	if(!do_after(user, 3 SECONDS, src))
		return
	to_chat(user, span_notice("The contacts seamlessly merge with your iris."))
	eyes.flash_protect = FLASH_PROTECTION_WELDER
	to_chat(user, span_warning("\The [src] disintegrates into nothing."))
	qdel(src)

```

**Comment by the-orange-cow** [comment]
I don't know how to read code, by single use, do you mean the contacts can be used on a character a single time or that they only block one flash before needing to buy another set?

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -397,6 +397,29 @@
 	inhand_icon_state = "gar"
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/syndicate_contacts
+	name = "suspicious contact case"
+	desc = "A sinister red case that contains two shiny black contancts lenses."
```
Comment (Sealed101) [inline code/modules/clothing/glasses/_glasses.dm:402]:
```suggestion
	name = "suspicious contact lens case"
	desc = "A sinister red case that contains two shiny black contact lenses."
```

File: code/modules/clothing/glasses/_glasses.dm [hunk 3]
```diff
@@ -397,6 +397,29 @@
...
+	w_class = WEIGHT_CLASS_TINY
+	icon = 'icons/obj/device.dmi'
+	icon_state = "contacts"
+
+/obj/item/syndicate_contacts/attack_self(mob/user, modifiers)
+	. = ..()
+	if(!user.get_organ_slot(ORGAN_SLOT_EYES))
+		to_chat(user, span_warning("You have no eyes to apply the contacts to!"))
```
Comment (Sealed101) [inline code/modules/clothing/glasses/_glasses.dm:410]:
```suggestion
	if(!user.get_organ_slot(ORGAN_SLOT_EYES))
		to_chat(user, span_warning("You have no eyes to apply the contacts to!"))
		return
```

**Review by Sealed101 — CHANGES_REQUESTED** [review]
this is much more powerful for moths and flypeople, as it sets the welding protection directly instead of buffing it. was this intended?

also having some clues about it on maybe like an advanced health scanner or something wouldn't hurt, otherwise these eyes are completely undetectable. lose your eyes to the local eye fridge and you won't find 'em easily

**Comment by TheVekter** [comment]
I'm liking this trend of giving antags answers to other antags. Cool item, hope the issues get ironed out.

**Comment by itseasytosee** [comment]
> Can it be seen via health analyzer / eyes removal? 

No, but they are cleared on eye removal.

**Comment by itseasytosee** [comment]
> this is much more powerful for moths and flypeople, as it sets the welding protection directly instead of buffing it. was this intended?
> 
> 
> 
> also having some clues about it on maybe like an advanced health scanner or something wouldn't hurt, otherwise these eyes are completely undetectable. lose your eyes to the local eye fridge and you won't find 'em easily

It would be a one character change to turn this to maxing out an eye's flash protection to just giving it a buff. I suppose thats probably better anyways.

**Comment by Sealed101** [comment]
yeah no making it a dead give-away is boring, maybe some reaction on the eyes as an item? maybe a subtle tell on how oculine seems to coat the eyeball around the iris? (does oculine work like that even i forget)

**Comment by jughu** [comment]
make the syndicate bible anti magic so its a answer in cult(and you still need to suit storage or hold in hand for effects)

**Comment by Sealed101** [comment]
wrong PR eden

**Comment by itseasytosee** [comment]
> make the syndicate bible anti magic so its a answer in cult(and you still need to suit storage or hold in hand for effects)

Already ahead of you #76733

**Comment by itseasytosee** [comment]
> yeah no making it a dead give-away is boring, maybe some reaction on the eyes as an item? maybe a subtle tell on how oculine seems to coat the eyeball around the iris? (does oculine work like that even i forget)

Is the idea that you would like traitors to be able to re-find their eyeballs if they happen to loose them or as a way to check if a traitor's eyes are lensed.

**Comment by itseasytosee** [comment]
contacts
It's a codersprite, so not very impressive.

**Comment by Sealed101** [comment]
> Is the idea that you would like traitors to be able to re-find their eyeballs if they happen to loose them or as a way to check if a traitor's eyes are lensed.

both probably, but the penlight would also work for welding eyes, which are a thing now that I've remembered it. up to you I guess

**Comment by jughu** [comment]
though ultimately i dont think the flash protection is one needed since theres alot of ways to get sunglasses like walking to brig for implant+gear since thats gauranteed to happen in revs theres some extra gar glasses in theatre vendors theres one in locker room maint and one in theatre

**Comment by HoodMilk** [comment]
Why not just have a pair of sunglasses purchasable? Or a single use autosurgeon of welding shield eyes?  Traitors get their power from tangible items that they have (Even implants can be found with a surgery, and sleeping carp can be found with syringes.) this sort of impossible to detect passive buff sticks out like a sore thumb compared to the rest of the traitor kit. It's like if traitors could purchase gas immunity or bonus armor that can't be found in any way, it's just bizarre.

Lings can get flash protection innately, but their entire shtick is organic abilities that can't be detected, and they have to give up X-ray for it. 

If a pair of sunglasses is too detectable for some reason you could just make a pair of chameleon sunglasses in the same way that chameleon no-slips exist.

**Comment by ATH1909** [comment]
should probably be less than 3 tc, since this becomes much less relevant as soon as you gank your first officer/raid your first locker. I do like it, though.

any plans for an ear protection item?

**Comment by itseasytosee** [comment]
> Why not just have a pair of sunglasses purchasable? Or a single use autosurgeon of welding shield eyes?  Traitors get their power from tangible items that they have (Even implants can be found with a surgery, and sleeping carp can be found with syringes.) this sort of impossible to detect passive buff sticks out like a sore thumb compared to the rest of the traitor kit. It's like if traitors could purchase gas immunity or bonus armor that can't be found in any way, it's just bizarre.
> 
> 
> 
> Lings can get flash protection innately, but their entire shtick is organic abilities that can't be detected, and they have to give up X-ray for it. 
> 
> 
> 
> If a pair of sunglasses is too detectable for some reason you could just make a pair of chameleon sunglasses in the same way that chameleon no-slips exist.

martial arts, holopara, and implants come to minds as other exeception

Chameleons sunglasses could be a good alternative though.

**Comment by itseasytosee** [comment]
> should probably be less than 3 tc, since this becomes much less relevant as soon as you gank your first officer/raid your first locker. I do like it, though.
> 
> 
> 
> any plans for an ear protection item?

Consider the fact they are more descrete than sunglasses, and can be paired with other huds.

I would imagine a chameleon bowmans headset would be the soundproofing item, but some other headset gimmick could be cool.

**Comment by jughu** [comment]
sunglasses is better you want revs to have visual indictators since early they want stealth to build up numbers and a traitor with the cheal 2 polarized eyes could fuck you up even welding goggles would turn your eyes white which is also a visual indictator

File: code/modules/clothing/glasses/_glasses.dm [hunk 4]
```diff
@@ -397,6 +397,28 @@
 	inhand_icon_state = "gar"
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/syndicate_contacts
```
Comment (Time-Green) [inline code/modules/clothing/glasses/_glasses.dm:400]: Please autodoc new types

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Pull request merged**